docs/pattern-matching.md

```diff
@@ -1,229 +1,366 @@
 ---
-title: Pattern Matching!
+title: Pattern Matching
 ---
 
-_Make sure you've read up on [Variant](variant.md) first_.
+_Quick overview: [Pattern matching](overview.md#pattern-matching)_
 
-**We're finally here**! Pattern matching is one of _the_ best features of the language. It's like destructuring, but comes with even more help from the type system.
+Pattern matching provides a way to conditionally execute code when the shape of some data matches a particular pattern. It is similar to switch-case statements in other languages, but it can be more expressive and includes some extra safeguards.
 
-## Usage
+Pattern matching is often used with [variants](variant.md).
 
-Consider a variant:
+## Defining a pattern match statement
+
+Use the `switch` keyword to make a pattern matching statement:
 
 ```reason
-type payload =
-  | BadResult(int)
-  | GoodResult(string)
-  | NoResult;
+switch (input) {
+| Pattern1 => code1
+| Pattern2 => code2
+| Pattern3 => code3
+};
 ```
 
-While using the `switch` expression on it, you can "destructure" it:
+The switch finds the first pattern that matches the input, and then executes the code that the pattern points to (the code after the `=>`). Code for a pattern can be a single expression, or a block of statements.
```

Comment:
Worth mentioning right up front that the `switch` evaluates to the value, and is a valid expression. (That's different than what JS users are accustomed to).